src/uucore/src/lib/features/format/num_format.rs

```diff
@@ -293,13 +293,7 @@ impl Formatter for Float {
 
         let precision = match precision {
             Some(CanAsterisk::Fixed(x)) => x,
-            None => {
-                if matches!(variant, FloatVariant::Shortest) {
```

Comment:
Hmm good catch. I was wondering how this came to be and it seems to be this: https://github.com/uutils/coreutils/commit/4c5326ffa34b2026d5437b8b5b118f5f67467f0a

Looks like that commit only looked at `%g` and check the other formats.

Reply:
Yeah... Also... Default precision for `%a` is not 6, but... the hexadecimal printing function right now does not support any sort of precision setting (e.g. `%.6a` still prints all digits...).

I'm... slowly trying to untangle this, maybe better to fix #7364 first. Still filed #7429 to remember.